daffodil-test/src/test/scala/org/apache/daffodil/runtime1/layers/AISTransformer.scala

```diff
@@ -20,130 +20,78 @@ package org.apache.daffodil.runtime1.layers
-        )
-    }
-    layerCompileInfo.optLayerLengthKind match {
-      case Some(LayerLengthKind.BoundaryMark) => // ok
-      case Some(other) =>
-        layerCompileInfo.SDE(
-          s"Only dfdlx:layerLengthKind 'boundaryMark' is supported, but '$other' was specified",
-        )
-      case None => // ok
-    }
-    layerCompileInfo.optLayerJavaCharsetOptConstantValue match {
-      case Some(Some(_)) => // ok
-      case None => // ok
-      case _ =>
-        layerCompileInfo.SDE(
-          "Property dfdlx:layerEncoding must be defined, and must be an ordinary 8-bit wide encoding. ",
-        )
-    }
-    val xformer = new AISPayloadArmoringTransformerFactory(name)
-    xformer
-  }
-}
-
-class AISPayloadArmoringTransformerFactory(name: String) extends LayerTransformerFactory(name) {
-
-  override def newInstance(layerRuntimeInfo: LayerRuntimeInfo) = {
-    val xformer = new AISPayloadArmoringTransformer(name, layerRuntimeInfo)
-    xformer
-  }
-}
-
-object AISPayloadArmoringTransformer {
-  def iso8859 = StandardCharsets.ISO_8859_1
-}
-
-class AISPayloadArmoringTransformer(name: String, layerRuntimeInfo: LayerRuntimeInfo)
-  extends LayerTransformer(name, layerRuntimeInfo) {
-
-  import AISPayloadArmoringTransformer._
+final class AISPayloadArmoringLayer
+  extends Layer(
+    layerName = "aisPayloadArmor",
+    supportedLayerLengthKinds = Seq(JLayerLengthKind.BoundaryMark).asJava,
+    supportedLayerLengthUnits = Seq().asJava,
+    isRequiredLayerEncoding = true,
+    optLayerVariables = Optional.empty(),
+  ) {
 
   /**
    * Decoding AIS payload armoring is encoding the ASCII text into the
    * underlying binary data.
    */
-  override def wrapLayerDecoder(jis: java.io.InputStream) = {
+  override def wrapLayerDecoder(jis: InputStream, lr: LayerRuntime): InputStream =
     new AISPayloadArmoringInputStream(jis)
-  }
-
-  override def wrapLimitingStream(state: ParseOrUnparseState, jis: java.io.InputStream) = {
-    val layerBoundaryMark = ","
-    val s = BoundaryMarkLimitingStream(jis, layerBoundaryMark, iso8859)
-    s
-  }
 
-  override protected def wrapLayerEncoder(jos: java.io.OutputStream): java.io.OutputStream = {
+  override def wrapLayerEncoder(jos: OutputStream, lr: LayerRuntime): OutputStream =
     new AISPayloadArmoringOutputStream(jos)
-  }
 
-  override protected def wrapLimitingStream(
-    state: ParseOrUnparseState,
-    jos: java.io.OutputStream,
-  ) = {
-    val layerBoundaryMark = ","
-    val newJOS = new LayerBoundaryMarkInsertingJavaOutputStream(jos, layerBoundaryMark, iso8859)
-    newJOS
-  }
+  override def layerLimiter(layerPropertyInfo: LayerPropertyInfo): Optional[LayerLimiter] =
+    Optional.of(new BoundaryMarkRegexLimiter {
+      override protected def regexForBoundaryMarkMatch: String = ","
+
+      override protected def boundaryMarkToInsert: String = ","
+
+      override protected def maximumLengthBoundaryMark: String = ","
+    })
```

Comment:
The above is just unnecessary. This is just regular boundaryMark behavior, where the boundary mark is a constant built into the layer code. Why bother. Just in the AIS layer DFDL file, define dfdlx:boundaryMark="," and then drop this stuff.